curriculum/challenges/english/25-front-end-development/lab-user-configuration-manager/684aaf9ec670c68d20efd0d0.md

```diff
@@ -0,0 +1,379 @@
+   - Convert the key to lowercase.
+   - If the key setting exists, return `Setting [key] already exists! Cannot add a new setting with this name.`
+   - If the key setting doesn't exist, add the key-value pair to the `settings` dictionary and return `Setting [key] added with value [value] successfully!`
+
+1. You should define a function named `update_setting` that takes the a dictionary of settings and a tuple containing a key-value pair as parameters.
+
+1. `update_setting` function should:
+       
+   - Unpack the tuple into a key and value.
+   - Convert the key to lowercase.
+   - If the key setting exists, update its value in the `settings` dictionary and return: `Setting [key] updated to [value] successfully!`
+   - If the key setting doesn't exist, return `Setting [key] does not exist! Cannot update a non-existing setting.`
+
+1. You should define a function named `delete_setting` that takes the a dictionary of settings and a key as parameters.
+
+1. `delete_setting` function should:
+    
+   - Convert the key passed to lowercase.
+   - If the key setting exists, remove the key-value pair from `settings` and return `Setting [key] deleted successfully!`
+   - If the key setting does not exist, return `Setting not found!`
+
+1. You should define a function named `view_settings` that takes a dictionary of settings as argument.
+
+1. `view_settings` function should:
+   
+   - Return `No settings available.` if the `settings` dictionary is empty. 
+   - If the dictionary contains any settings, return them formatted as follows:
+
+   ```md
+   Current User Settings:
+   Theme: dark
+   Notifications: enabled
+   Volume: high
+   ```
+
+1. For testing the code, you should create a dictionary named `test_settings` that would store the following user configuration preferences in this format:
+
+   ```md
+    "theme": "light",
+    "language": "English",
+    "notifications": "enabled"
+   ```
+
+# --hints--
+
+You should create a dictionary named `test_settings` and add some values to it.
+
+```js
+({
+    test: () => runPython(`
+        length = len(test_settings)
+        assert length > 0
+    `)
+})
+```
+
+You should define a function named `add_setting`.
+
+```js
+({ test: () => assert(runPython(`_Node(_code).has_function("add_setting")`)) })
+```
+
+The `add_setting` function should take two parameters.
+
+```js
+({
+    test: () => runPython(`
+        import inspect
+        sig = inspect.signature(add_setting)
+        assert len(sig.parameters) == 2
+    `)
+})
+```
+
+`add_setting` should convert the key to lowercase.
+
+```js
+({
+    test: () => runPython(`
+        settings_in_test = {"theme": "light"}
+        result = add_setting(settings_in_test, ("VOLUME", "high"))
+        assert "volume" in settings_in_test
+        assert "VOLUME" not in settings_in_test
+    `)
+})
+```
+
+`add_setting` should return the error message `Setting [key] already exists! Cannot add a new setting with this name.` when the key already exists.
```

Comment:
The expected message seems to be different from the one shown. I'm referring to the quotes around the key. This happens for other messages too.

Reply:
Changed `[key]` to `'[key]'` and `[value]` to `'[value]'` where applicable.